commit/merkleroot/processor.go

```diff
@@ -0,0 +1,79 @@
+package merkleroot
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/plugincommon"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
+	"github.com/smartcontractkit/chainlink-ccip/shared"
+)
+
+type Processor struct {
+	nodeID       commontypes.OracleID
+	cfg          pluginconfig.CommitPluginConfig
+	lggr         logger.Logger
+	observer     Observer
+	readerSyncer *plugincommon.BackgroundReaderSyncer
+	ccipReader   reader.CCIP
```

Comment:
I believe it's better if we move the plugin methods in this file.
Then we have:
```
merkleroot/
   processor.go
   processor_test.go
   types.go
   types_test.go
```

Right now I find it hard to navigate through the different files.

Reply:
I think it's more readable having different files for observation, outcome, report,..etc. I can change the other processors to follow the same scheme so that navigating files is easier. @makramkd @winder WDYT? In all cases I don't think this needs to be done in this PR. We want to unblock the work on tokens and fees for the time being.

Reply:
Different strokes for different folks, I like the locality as long as the file is < 1,000 lines.

Reply:
I like locality if it's < 1000 lines, this one will be above the 1K lines and the tests specifically will be hard to navigate as it will be much more than 1K lines.

Reply:
I don't have a strong preference, but usually lean towards keeping an interface implementation all in one file. That said execute is huge and I was planning to split it up like this at some point.

Either way I agree that it should be done in a separate PR.